src/components/shared/ButtonWithPanel.js

```diff
@@ -2,15 +2,43 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React, { PureComponent, PropTypes } from 'react';
+// @flow
+
+import React, { PureComponent } from 'react';
 import classNames from 'classnames';
 
 require('./ButtonWithPanel.css');
 
+interface Panel {
+  props: {
+    onOpen: () => mixed,
+    onClose: () => mixed,
+  },
+  open(): mixed,
+}
+
+type Props = {
+  className: string,
+  label: string,
+  panel: React$Element<*>, // Ideally we'd like to say that panel implements Panel, but I can't express it with Flow
```

Comment:
Does it not work if you pass the Class here? Either way this is probably good enough.

Reply:
I think I tried everything :) `Panel` is an interface, that's why it doesn't work like this... Maybe with the new React stuff in Flow v0.54...